test/e2e/framework/metrics/metrics_grabber.go

```diff
@@ -53,26 +54,31 @@ type Collection struct {
 type Grabber struct {
 	client                            clientset.Interface
 	externalClient                    clientset.Interface
+	config                            *rest.Config
 	grabFromAPIServer                 bool
 	grabFromControllerManager         bool
 	grabFromKubelets                  bool
 	grabFromScheduler                 bool
 	grabFromClusterAutoscaler         bool
 	kubeScheduler                     string
-	waitForSchedulerReadyOnce         sync.Once
 	kubeControllerManager             string
-	waitForControllerManagerReadyOnce sync.Once
```

Comment:
I removed the `sync.Once` check because:
- in practice it was always done because the Grabber instances only seem to be used once (otherwise https://github.com/kubernetes/kubernetes/commit/c496b1d335862cef202753d815323ce884dac862 would have broken tests)
- if called more than once, checking for readiness repeatedly shouldn't cause much overhead (pods already ready) and may be useful (in the unlikely case that pods restart)